widget/text.go

```diff
@@ -341,11 +341,57 @@ func splitLines(text []rune) [][2]int {
 	return append(lines, [2]int{low, length})
 }
 
+// newBinarySearch accepts a function that checks if the text width between the start and end index less the maximum width
+// newBinarySearch returns a function that accepts the start and end rune index and returns the index of rune
+// located as close to the maximum line width as possible
+func newBinarySearch(lessMaxWidth func(int, int) bool) func(low int, maxHigh int) int {
```

Comment:
I'm not sure this needs to be a func that returns a func. Why couldn't this be binarySearch(func, int, int) int?

Rather than:

```
binarySearch := newBinarySearch(func(...
...
binarySearch(low, high)
```

It would be:

```
ruler := func(...
...
binarySearch(ruler, low, high)
```

I think it would look cleaner, but am I missing something?